monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -175,46 +176,71 @@ def _create_cmd(self, train_params=None):
+            else:
+                cmd = 'python ' if cmd is None else cmd
         cmd += base_cmd
         if params and isinstance(params, Mapping):
             for k, v in params.items():
                 cmd += f" --{k}={v}"
-        return cmd, devices_info
+        return cmd
 
-    def _run_cmd(self, cmd: str, devices_info: str) -> subprocess.CompletedProcess:
+    def _run_cmd(self, cmd: str) -> subprocess.CompletedProcess:
         """
         Execute the training command with target devices information.
 
         """
 
         logger.info(f"Launching: {cmd}")
         ps_environ = os.environ.copy()
-        if devices_info:
-            ps_environ["CUDA_VISIBLE_DEVICES"] = devices_info
-        normal_out = subprocess.run(cmd.split(), env=ps_environ, check=True)
-
+        ps_environ["CUDA_VISIBLE_DEVICES"] = self.device_setting['CUDA_VISIBLE_DEVICES']
+        if self.device_setting['NUM_NODES'] > 1:
+            if self.device_setting['MN_START_METHOD'] == 'bcprun':
+                normal_out = subprocess.run(['bcprun','-n',str(self.device_setting['NUM_NODES']),'-p',
+                                             str(self.device_setting['n_devices']),'-c', cmd], env=ps_environ, check=True)
+            else:
+                raise NotImplementedError(f"{self.device_setting['MN_START_METHOD']} is not supported yet.\
+                                             Try modify BundleAlgo._run_cmd for your cluster.")
+        else:
+            normal_out = subprocess.run(cmd.split(), env=ps_environ, check=True)
         return normal_out
 
-    def train(self, train_params=None):
+    def train(self, train_params: None | dict = None, device_setting: None | dict = None):
         """
         Load the run function in the training script of each model. Training parameter is predefined by the
         algo_config.yaml file, which is pre-filled by the fill_template_config function in the same instance.
 
         Args:
-            train_params:  to specify the devices using a list of integers: ``{"CUDA_VISIBLE_DEVICES": [1,2,3]}``.
+            train_params:  training parameters
+            device_settings: device related settings, should follow the device_setting in auto_runner.set_device_info.
+            'CUDA_VISIBLE_DEVICES' should be a string e.g. '0,1,2,3'
         """
-        cmd, devices_info = self._create_cmd(train_params)
-        return self._run_cmd(cmd, devices_info)
+        # device_setting set default value and sanity check, in case device_setting not from autorunner
+        self.device_setting = {'CUDA_VISIBLE_DEVICES': ','.join([str(x) for x in range(torch.cuda.device_count())]),
+                               'n_devices': torch.cuda.device_count(), 'NUM_NODES': int(os.environ.get('NUM_NODES', 1)),
+                               'MN_START_METHOD': os.environ.get('MN_START_METHOD', 'bcprun'),
+                               'CMD_PREFIX':os.environ.get('CMD_PREFIX', None),
```

Comment:
`CMD_PREFIX` seems a broad concept. I feel confused what may be put there.

Reply:
wenqi suggested user defined runner
export CMD_PREFIX="python "
export CMD_PREFIX="/opt/conda/bin/python3.8 "
export CMD_PREFIX="torchrun --nnodes=3 --nproc_per_node=2 "

Reply:
The examples are helpful for me. Maybe we can put that in the doc-string?